<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PointSetShapeDetectionDialog</class>
 <widget class="QDialog" name="PointSetShapeDetectionDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>495</width>
    <height>552</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Shape Detection</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Algorithm</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QRadioButton" name="m_region_growing">
        <property name="text">
         <string>Region growing</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QRadioButton" name="ransac">
          <property name="text">
           <string>RANSAC</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="planeCB">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Plane</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="cylinderCB">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Cylinder</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="torusCB">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Torus</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="coneCB">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Cone</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="sphereCB">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Sphere</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="m_regularize">
        <property name="text">
         <string>Regularize Planes</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="m_generate_structured">
        <property name="text">
         <string>Generate structured point cloud</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Parameters</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Epsilon:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="DoubleEdit" name="m_epsilon_field">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>0.002</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Normal Tolerance (in degrees):</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="DoubleEdit" name="m_normal_tolerance_field">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>30</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,0">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Minimum Number of Items:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="m_min_pts_field">
          <property name="toolTip">
           <string>Smallest allowed size for a primitive</string>
          </property>
          <property name="maximum">
           <number>10000</number>
          </property>
          <property name="singleStep">
           <number>50</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,0">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Connectivity Epsilon:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="DoubleEdit" name="m_cluster_epsilon_field">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>0.002</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="0,0">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Search Rigorosity (Probability):</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="DoubleEdit" name="m_probability_field">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>0.05</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Output</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QCheckBox" name="m_add_property">
        <property name="text">
         <string>Add a &quot;shape&quot; property to the input item</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="m_one_colored_point_set">
        <property name="text">
         <string>Generate one colored point set</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="m_point_subsets">
        <property name="text">
         <string>Generate N point subsets</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="m_alpha_shapes">
        <property name="text">
         <string>Generate N alpha shapes (planes only)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DoubleEdit</class>
   <extends>QLineEdit</extends>
   <header>CGAL_double_edit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PointSetShapeDetectionDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PointSetShapeDetectionDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ransac</sender>
   <signal>toggled(bool)</signal>
   <receiver>planeCB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>170</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ransac</sender>
   <signal>toggled(bool)</signal>
   <receiver>cylinderCB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ransac</sender>
   <signal>toggled(bool)</signal>
   <receiver>torusCB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ransac</sender>
   <signal>toggled(bool)</signal>
   <receiver>coneCB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ransac</sender>
   <signal>toggled(bool)</signal>
   <receiver>sphereCB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ransac</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_5</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>103</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
